server-spi/src/main/java/org/keycloak/storage/changeset/UserData.java

```diff
@@ -104,6 +108,7 @@ public boolean isChanged() {
         return !idChanged
```

Comment:
negation makes this expession rather complicated...
why not 
```return idChanged || usernameChanged || ....``` 
Shold the updatedTimestampChanged be a part of that change detection at all? since it is a derivedProperty...